text/0118-search-api.md

```diff
@@ -426,9 +431,146 @@ Sets the maximum number of documents to be returned for the search query.
+
+- 🔴 Sending a value with a different type than `Integer` for `hitsPerPage` returns a [bad_request](0061-error-format-and-definitions.md#bad_request) error.
+
+#### 3.1.8.1. Navigating search results by page selection
+
+By default, `limit` and `offset` are used for pagination. That pagination system, while performant, does not provide the required information to create a correct page selection. Upon using `limit`/`offset`, `estimatedTotalHits` is returned which provides a rough estimation of how many hits may be candidates for a given request. See [`limit`/`offset` usage](#31811-limitoffset-usage) for further explaination.
+
+The `page selection` system provides an alternative that tackles the above-mentioned issue when the user needs reliable information for its pagination. For example, when creating a pagination with numbers `<< < 1, 2, 3, ...14 > >>`. Nonetheless, it is less performant as the engine needs to compute the `totalHits` exhaustively. 
+With this page selection system, it is possible to jump from one page to another using the `page` parameter and decide how many hits should be present in a page with `hitsPerPage`. See [`page`/`hitsPerPage` usage](#31812-pagehitsperpage-usage) for further explaination.
+
+##### 3.1.8.1.1. Limit/offset usage
+
+When either `limit` or `offset` is specified or when neither `limit`, `offset`, `page` and `hitsPerPage` are specific, the response object contains related fields:
+- [`estimatedTotalHits`](#324-estimatedtotalhits)
+- [`limit`](#322-limit)
+- [`offset`](#323-offset)
+
+If in addition to `limit` and/or `offset`, either `page` or `hitsPerPage` is also provided, `limit` and `offset` parameters are ignored.
+
+For example, given the following query parameters:
+
+- limit: 10
+- offset: 1
+
+The response objects contain these specific fields:
+```json
+{
+    "hits": [
+        /// ... 10 hits
+    ],
+    /// ... other fields
+    "limit": 10,
+    "offset": 1,
+    "estimatedTotalHits": 1345
+}
+```
+
+For example, on a query with no query parameters:
+
+The response objects contain these specific fields:
+```json
+{
+    "hits": [
+        /// ... 10 hits
+    ],
+    /// ... other fields
+    "limit": 20,
+    "offset": 0,
+    "estimatedTotalHits": 1345
+}
+```
+
+##### 3.1.8.1.2 page/hitsPerPage usage
+
+As soon as either `page` or `hitsPerPage` is used as a query parameter, in the response object, `limit`, `offset`, and `estimatedTotalHits` are removed and new fields are returned:
+
+- [`hitsPerPage`](#326-hitsperpage): number of results in each search results page.
+- [`page`](#325-page): current search results page. The counting starts at 1.
+- [`totalPages`](#327-totalpages): total number of results pages. Calculated using `hitsPerPage` value.
+- [`totalHits`](#328-totalhits): total number of search results.
+
+Both `totalPages` and `totalHits` are computed until they reach the `pagination.maxTotalHits` number from the settings. Default (1000).
```

Comment:
Could be nice to have a link to the maxTotalHits parameter in the pagination settings specs here 🤓 

Removing the Default (1000) prevents us from maintaining it at multiple places if we change it in the future. WDYT?

Reply:
```suggestion
Both `totalPages` and `totalHits` are computed until they reach the [`pagination.maxTotalHits`](#https://github.com/meilisearch/specifications/blob/main/text/157-pagination-setting-api.md#311-maxtotalhits) number from the settings.
```